### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by Adakar**
Documentation will be build on demand from the main branch. After that it's going to be deployed to GH pages.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+```sh
+pip install squape
+```
+
+### Content
+The package consists of several modules:
+- report - adds features to enhance the reporting capabilities of Squish
+- settings - makes using various test settings easier and cleaner
+- video - adds features to enhance video capture capabilities of Squish
+- vps - extension of Squish verification points
```
Comment (pawtom) [inline docs/index.md:22]: object_tree module should be mentioned here as well

**Review by pawtom — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 1 above)
Comment (Adakar) [reply docs/index.md:24]: added :)

File: docs/reference/object_tree.md [hunk 2]
```diff
@@ -0,0 +1 @@
+::: squape.object_tree
```
Comment (EduardKaverinskyi) [inline docs/reference/object_tree.md:1]: What about filters for some of the pages to remove private methods?

options:
        filters: ["!^_"]

File: squape/object_tree.py [hunk 3]
```diff
@@ -265,7 +263,7 @@ def _get_object_reference(object_name: any) -> any:
         object_name (any): symbolic name, real name, or object reference.
 
     Returns:
-        object_reference: Object reference from the given object_name.
+        (Squish object): Object reference from the given object_name.
```
Comment (EduardKaverinskyi) [inline squape/object_tree.py:266]: Will those changes be present in the sources too?

File: squape/report.py [hunk 4]
```diff
@@ -216,8 +216,9 @@ def fatal(msg: str, details: str = "") -> None:
 def enable_loglevel_in_test_module() -> None:
     """Adds support for log levels to the Squish 'test' module.
 
-    DISCLAIMER: This function uses monkeypathching
-    https://en.wikipedia.org/wiki/Monkey_patch
+    !!! warning
+        This function uses monkey pathching
+        https://en.wikipedia.org/wiki/Monkey_patch
```
Comment (EduardKaverinskyi) [inline squape/report.py:221]: Will those changes be present in the sources too?

File: squape/settings.py [hunk 5]
```diff
@@ -57,7 +57,7 @@ def logScreenshotOnPass(enabled: bool = True) -> None:
 
     Args:
         enabled (bool): A boolean value indicating whether to enable logging
-        of screenshots on PASS test result. Defaulting to True.
+            of screenshots on PASS test result. Defaulting to True.
```
Comment (EduardKaverinskyi) [inline squape/settings.py:60]: Will those changes be present in the sources too?

File: squape/video.py [hunk 6]
```diff
@@ -63,7 +63,7 @@ def _replace_videos(videos: set) -> None:
 
     Args:
         videos (set): A set of video names to replace.
-        Does nothing when the set is empty.
+            Does nothing when the set is empty.
```
Comment (EduardKaverinskyi) [inline squape/video.py:66]: Will those changes be present in the sources too?

File: squape/object_tree.py (hunk 3 above)
Comment (Adakar) [reply squape/object_tree.py:266]: I don't understand the question.

File: squape/report.py (hunk 4 above)
Comment (Adakar) [reply squape/report.py:221]: I don't understand the question.

File: squape/settings.py (hunk 5 above)
Comment (Adakar) [reply squape/settings.py:60]: I don't understand the question.

File: squape/video.py (hunk 6 above)
Comment (Adakar) [reply squape/video.py:66]: I don't understand the question.

File: squape/object_tree.py (hunk 3 above)
Comment (EduardKaverinskyi) [reply squape/object_tree.py:266]: I meant that would those changes to the docstrings also apply in the original files or will only remain in the documentation? I had a second look and it looks like those changes would be universal, so it's ok.

**Review by pawtom — CHANGES_REQUESTED** [review]

File: squape/object_tree.py [hunk 7]
```diff
@@ -159,12 +157,12 @@ def height_filter_function(height: int) -> bool:
...
 
-def siblings(object_name: any, selector: dict = None):
+def siblings(object_name: any, selector: dict = None) -> tuple:
     """
-    Find all the object's siblings.
+    Find the object's siblings.
 
     Args:
-        object_name (any): symbolic name, real name, or object reference to the parent.
+        object_name (any): symbolic name, real name, or object reference.
```
[the remark this answers is not in the document]
Comment (Adakar) [reply squape/object_tree.py:165]: objectName is the parameter used across most if not all of the Squish lookup functions. The only change was to use snake_case convention.Changing it, and staying consistent would require changes in the vps module.

This variable was not introduced or modified in this PR , if you would like changing it, please open a discussion, issue or PR.

**Review by pawtom — APPROVED** [review]

**Pull request merged**